src/include/duckdb/planner/table_binding.hpp

```diff
@@ -101,7 +104,7 @@ struct TableBinding : public Binding {
 public:
 	TableBinding(const string &alias, vector<LogicalType> types, vector<string> names,
 	             vector<ColumnIndex> &bound_column_ids, optional_ptr<StandardEntry> entry, idx_t index,
-	             bool add_row_id = false);
+	             bool add_row_id = false, LogicalType rowid_type = LogicalType(LogicalType::ROW_TYPE));
```

Comment:
`TableBinding` already takes a StandardEntry (which should be a `TableCatalogEntry` if `add_row_id` is true). Can we just call `entry->Cast<TableCatalogEntry>().GetRowIdType()` instead to avoid having to pull this through many layers?